app/src/main/java/com/kadequart/android/quartrack/Transaction.java

```diff
@@ -24,4 +25,15 @@ public double getAmount () {
     public void setAmount (double amount) {
         this.amount = amount;
     }
+
+    static int generateNextId (Realm realm) {
+        Number size = realm.where(Transaction.class).max("id");
+
+        if (size != null) {
+            return size.intValue() + 1;
+        }
+
+        return 1;
+    }
+
```

Comment:
Better implementation, remove from transaction object, then create a utility class with this function:
```
  public static int getNextId(Realm realm, Class classToSearch) {
    if (realm.where(classToSearch).max("id") != null) {
      return Integer.parseInt(realm.where(classToSearch).max("id").toString()) + 1;
    }

    return 1;
  }
```